File: Client/types/notification.ts
export enum NotificationType {
  TaskAssigned = 1,
  TaskSubmitted = 2,
  TaskApproved = 3,
  ChapterSubmittedForReview = 4,
  ChapterApproved = 5,
  RankingCalculated = 6,
  CancellationWarning = 7,
  FileUploaded = 8,
  System = 9,
  ChapterRevisionRequested = 10,
  ChapterRejected = 11,
  ReaderChapterPublished = 12,
  ProposalSubmitted = 13,
  ProposalApproved = 14,
  ProposalRejected = 15,
  CancellationWarningCreated = 16,
  SeriesCancelled = 17,
  SeriesHiatus = 18,
}

export enum NotificationStatus {
  Unread = 1,
  Read = 2,
}

export interface NotificationResponse {
  id: string;
  userId: string;
  title: string;
  message: string;
  type: NotificationType;
  status: NotificationStatus;
  sourceEventType?: string;
  sourceEventId?: string;
  resourceType?: string;
  resourceId?: string;
  actionUrl?: string;
  createdAt: string;
  readAt?: string;
}

export interface UnreadCountResponse {
  unreadCount: number;
}

File: Server/services/notification-service/Manga.Notification.Domain/Entities/Notification.cs
using Manga.Notification.Domain.Enums;

namespace Manga.Notification.Domain.Entities;

public sealed class Notification
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid UserId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Message { get; set; } = string.Empty;

    public NotificationType Type { get; set; } = NotificationType.System;

    public NotificationStatus Status { get; set; } = NotificationStatus.Unread;

    public string? SourceEventType { get; set; }

    public Guid? SourceEventId { get; set; }

    public string? ResourceType { get; set; }

    public Guid? ResourceId { get; set; }

    public string? ActionUrl { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? ReadAt { get; set; }
}
